### PR TITLE
chore:remove space b/w form and CTA onboarding page

### LINKED ISSUE #30523: Remove unnecessary space between form and the CTA for GSheets in the datasource review form (onboarding step)
Problem:
[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeC9wQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3e0f9af90f233e63ff74e34b84df071519574bf9/Unecessary%20space.mov)

Solution: 
- Remove the unnecessary space ([design](https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=5340-6425&mode=design&t=0EXdE30l5U090Tzc-0))

Additional improvements (also presented in the previous design link)
- Keep the title "Connect data" and supplementary test "Your data's security is our priority."
- Replace the current dropdown by radio buttons.
- Move the callout to after the permissions | scope property. 
- Limit the width of the white section

### PR DESCRIPTION
**Pull request by Shivam-z**
## Description

Following are the improvements made in this PR:

- Remove the unnecessary space b/w form and CTA in Gsheet onboarding step
- Made one new RadioButtonControl in form control and replaced the current 
dropdown by radio buttons.
- Move the callout to after the permissions | scope property.
- Limit the width of the white section

Fixes #30523

output screenshot:
Screenshot from 2024-09-20 15-14-59

Desired design:
image

>
> _Please also include relevant motivation and context. List any dependencies that are required for this change. Add links to Notion, Figma or any other documents that might be relevant to the PR._

*Fixes #`35950`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `RadioButtonControl` component for improved form control options.
	- Enhanced the `FormControlRegistry` to support radio button controls.
	- Updated the Google Sheets plugin to use radio buttons for permission settings.

- **UI Changes**
	- Corrected styling syntax in the `FormContainer` for proper rendering.
	- Reorganized the display order of information banners in the `DatasourceForm` for better clarity.

- **Tests**
	- Added a comprehensive suite of unit tests for the `RadioButtonControl` component to ensure proper functionality and user interaction.

### CODE REVIEW

**Comment by sagar-qa007** [comment]
@Shivam-z Should we consider updating an existing test case or adding a new one for unit testing?
Your insights would be appreciated.

**Comment by Shivam-z** [comment]
@sagar-qa007 we can add a jest unit testing for **RadioButtonControl**.

**Comment by Shivam-z** [comment]
@sagar-qa007 as discussed earlier, I have added a jest unit test case for **RadioButtonControl**.
can you please review it.

**Comment by ayushpahwa** [comment]
Tagging @AmanAgarwal041 @sneha122 for review

File: app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import type { FeatureFlags } from "ee/entities/FeatureFlag";
 export const FormContainer = styled.div`
   display: flex;
   position: relative;
-  height: 100%;
+  height: 55vh;
```
Comment (AmanAgarwal041) [inline app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx:15]: Why is this hard coded to `55vh` ? Will this not hinder with the responsiveness ? @Shivam-z

File: app/client/src/components/formControls/RadioButtonControl.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  const onChangeHandler = (value: string) => {
+    props.input && props.input.onChange && props.input.onChange(value);
+  };
+
+  const options = props.options || [];
+  const defaultValue = props.initialValue as string;
+
+  return (
+    <RadioGroup
+      data-testid={props?.input?.name}
```
Comment (AmanAgarwal041) [inline app/client/src/components/formControls/RadioButtonControl.tsx:41]: Why is this extra check for props required ? `props?.input?.name` -> `props.input?.name`

File: app/client/src/components/formControls/RadioButtonControl.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+type renderComponentProps = RadioButtonControlProps & {
+  input?: WrappedFieldInputProps;
+  meta?: WrappedFieldMetaProps;
+  options?: Array<{ label: string; value: string }>;
+};
+
+function renderComponent(props: renderComponentProps) {
+  const onChangeHandler = (value: string) => {
+    props.input && props.input.onChange && props.input.onChange(value);
```
Comment (AmanAgarwal041) [inline app/client/src/components/formControls/RadioButtonControl.tsx:33]: Try changing it to `typeof props.input?.onChange === "function"`, would this be a better way to check for onChange functionality ?

**Comment by carinanfonseca** [comment]
Hello @Shivam-z,
The spacing does not conform to the Appsmith Design System guidelines. There should be 16px of space between "Permissions | Scope" and the first radio option. For future reference, spacing should always be a multiple of 4px.

File: app/client/src/components/formControls/RadioButtonControl.tsx (hunk 2 above)
Comment (Shivam-z) [reply app/client/src/components/formControls/RadioButtonControl.tsx:41]: The height was originally hardcoded to eliminate the space between the form and the CTA buttons. But In the next commit, I removed the hardcoded value and adjusted the spacing by removing the height of the **DSEditorWrapper**.

File: app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx (hunk 1 above)
Comment (Shivam-z) [reply app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx:15]: The height was originally hardcoded to eliminate the space between the form and the CTA buttons. But In the next commit, I removed the hardcoded value and adjusted the spacing by removing the height of the **DSEditorWrapper**.

File: app/client/src/components/formControls/RadioButtonControl.tsx (hunk 3 above)
Comment (Shivam-z) [reply app/client/src/components/formControls/RadioButtonControl.tsx:33]: Yes, using **typeof props.input?.onChange === "function"** would be a better way to check if onChange is a function

File: app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx [hunk 4]
```diff
@@ -12,7 +12,7 @@ import type { FeatureFlags } from "ee/entities/FeatureFlag";
 export const FormContainer = styled.div`
   display: flex;
   position: relative;
-  height: 100%;
+  height 100%;
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx:15]: @Shivam-z Please look into the above changes suggested by coderabbit

**Comment by AmanAgarwal041** [comment]
> Hello @Shivam-z, The spacing does not conform to the Appsmith Design System guidelines. There should be 16px of space between "Permissions | Scope" and the first radio option. For future reference, spacing should always be a multiple of 4px.

@Shivam-z Please look into this as well.

File: app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Shivam-z) [reply app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx]: done

**Comment by AmanAgarwal041** [comment]
Hey @Shivam-z , Can you look into these cypress failures ?
Screenshot 2024-09-23 at 3 56 02 PM

**Comment by AmanAgarwal041** [comment]
@Shivam-z Hey, just a gentle follow up on [this](https://github.com/appsmithorg/appsmith/pull/35985#issuecomment-2367807512) ?

**Comment by Shivam-z** [comment]
@AmanAgarwal041 I have rebased my current branch with the release branch. 
**GoogleSheets_spec.ts** is running fine now.
can you please trigger the workflow one more time to verify the cypress results.

**Review by ichik — DISMISSED** [review]
Visual refinement is a definite improvement.

**Comment by AmanAgarwal041** [comment]
@Shivam-z There are still failures on the branch, can you please have a look at them ? 
Screenshot 2024-09-26 at 5 15 05 PM

**Comment by Shivam-z** [comment]
@AmanAgarwal041 I don't know why it is failing in the workflow run.
But it is running fine in my local , I am attaching a video for reference.

[Screencast from 26-09-24 05:26:59 PM IST.webm](https://github.com/user-attachments/assets/a2202f43-ad25-470a-90e2-fc47d9b3d27d)

**Comment by AmanAgarwal041** [comment]
@Shivam-z It should be fixed now for Google sheet. Other specs are failing though. Can you try others as well on the local ?

**Comment by KelvinOm** [comment]
@jsartisan could you please review this PR?

**Comment by Shivam-z** [comment]
@AmanAgarwal041 I’ve identified the root cause of the test case failures. The issue stemmed from the removal of the following height property in the **DSEditorWrapper**:

```css
height: calc(100vh - ${(props) => props.theme.headerHeight});
```

This height adjustment caused Cypress to fail in locating the required elements, leading to the test failures. In this [commit](https://github.com/appsmithorg/appsmith/pull/35985/commits/e5f4baba6b317f43eb54a61a89548b6b19230310), I’ve restored the original height value.

Could you please trigger the workflow again to verify the fix?`

**Comment by AmanAgarwal041** [comment]
Hey @Shivam-z , these are the test cases that are failing. Can you please check ?
Screenshot 2024-10-01 at 11 54 44 AM

**Comment by Shivam-z** [comment]
@AmanAgarwal041 Fixed GoogleSheet_specs file and rebased current branch with relase with latest changes.
All test cases are running in local.
can you trigger the workflow once again.

**Comment by AmanAgarwal041** [comment]
@Shivam-z There seems to be some lint errors. Can you fix those https://github.com/appsmithorg/appsmith/actions/runs/11130343332/job/31021794899?pr=35985 ?

**Comment by Shivam-z** [comment]
@AmanAgarwal041 I have fixed client linting errors.
please have look on it.

**Comment by AmanAgarwal041** [comment]
Here is one more linting error! @Shivam-z can you take a look at it ?
Screenshot 2024-10-08 at 12 31 18 AM

**Comment by Shivam-z** [comment]
@AmanAgarwal041 fixed above linting error

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**